### PR TITLE
refactor: ledger microservice APIs

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What?
- Modify GET API of `/platform/schemas` (get all schemas from platform).
- Modify GET API of `/platform/ledgers` (get all ledgers from platform).
- Modify GET API of `/platform/network/url/{indyNameSpace}` (get network URL from platform).
- Modify GET API of `/orgs/{orgId}/schemas/{schemaId}` (get schema information from the ledger by using schema Id).
- Modify GET API of `/orgs/{orgId}/schemas/{schemaId}/cred-defs` (get cred-defs by schema Id).
- Modify GET API of `/orgs/{orgId}/schemas` (get schemas by org Id).
- Modify  POST API of `/orgs/{orgId}/schemas` (create and sends a schema to the ledger).

1.Checked validations on payload.
2.Refactor response
3.Ensured consistency of API status codes.
4.Modified the NATS function.

### Why?
- The implementation of a well-defined interface enhances code readability and makes the structure of the API response more explicit.
- Ensuring consistent API status codes improves communication between different components of the system and enhances overall reliability.

### How?
Replaced the return type with a well-defined interface to enhance code clarity.
Checked and standardized API status codes to ensure a uniform response.
Modified the NATS function to accommodate the adjustments made in the API.

### CODE REVIEW

**Review by shitrerohit — APPROVED** [review]

File: apps/api-gateway/src/platform/platform.controller.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 import { Response } from 'express';
 import { ISchemaSearchPayload } from '../interfaces/ISchemaSearch.interface';
-import IResponseType, { IResponse } from '@credebl/common/interfaces/response.interface';
+import { IResponse } from '@credebl/common/interfaces/response.interface';
 import { ResponseMessages } from '@credebl/common/response-messages';
 import { CustomExceptionFilter } from 'apps/api-gateway/common/exception-handler';
 import { AuthGuard } from '@nestjs/passport';
 import * as QRCode from 'qrcode';
-
+import { SortFields } from '../schema/enum/schema.enum';
```
Comment (tipusinghaw) [inline apps/api-gateway/src/platform/platform.controller.ts:15]: any specific reason to use schema enum in platform service?

File: apps/api-gateway/src/schema/interfaces/index.ts [hunk 2]
```diff
@@ -39,4 +39,22 @@ export interface IOrgAgentInterface {
...
   agentsTypeId: string;
   orgId: string;
-}
\ No newline at end of file
+}
+
+export interface ISchemaInfo {
+  schema: Ischema;
+  schemaId: string;
+  resolutionMetadata: object;
```
Comment (tipusinghaw) [inline apps/api-gateway/src/schema/interfaces/index.ts:47]: If possiable we should create interface for `resolutionMetadata`

File: apps/ledger/src/repositories/ledger.repository.ts [hunk 3]
```diff
@@ -21,18 +22,16 @@ export class LedgerRepository {
         }
     }
 
-    async getNetworkUrl(payload: object): Promise<{
-        networkUrl: string;
-    }> {
+    async getNetworkUrl(indyNamespace: string): Promise<INetworkUrl> {
 
         try {
             return this.prisma.ledgers.findFirst({
```
Comment (tipusinghaw) [inline apps/ledger/src/repositories/ledger.repository.ts:28]:
```suggestion
            return this.prisma.ledgers.findFirstOrThrow({
```

Comment (tipusinghaw) [inline apps/ledger/src/repositories/ledger.repository.ts:28]:
```suggestion
            return this.prisma.ledgers.findFirstOrThrow({
```

File: apps/ledger/src/schema/repositories/schema.repository.ts [hunk 4]
```diff
@@ -165,7 +165,12 @@ export class SchemaRepository {
     try {
       const credDefResult = await this.prisma.credential_definition.findMany({
         where: {
-          AND: [{ orgId }, { schemaLedgerId: schemaId }]
+          AND: [{ orgId }, { schemaLedgerId: schemaId }],
+          OR: [
+            { tag: { contains: payload.searchByText, mode: 'insensitive' } },
```
Comment (tipusinghaw) [inline apps/ledger/src/schema/repositories/schema.repository.ts:170]: Please use variable destructuring

File: libs/common/src/response-messages/index.ts [hunk 5]
```diff
@@ -181,7 +181,8 @@ export const ResponseMessages = {
     },
     ledger: {
         success: {
-            fetch: 'Ledgers retrieved successfully.'
+            fetch: 'Ledgers retrieved successfully.',
```
Comment (tipusinghaw) [inline libs/common/src/response-messages/index.ts:184]: Please change the messages, we are not fecthing the ledger directly

File: apps/api-gateway/src/platform/platform.controller.ts (hunk 1 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/platform/platform.controller.ts:15]: Resolved

File: apps/ledger/src/repositories/ledger.repository.ts (hunk 3 above)
Comment (bhavanakarwade) [reply apps/ledger/src/repositories/ledger.repository.ts:28]: Resolved

File: libs/common/src/response-messages/index.ts (hunk 5 above)
Comment (bhavanakarwade) [reply libs/common/src/response-messages/index.ts:184]: Resolved

File: apps/ledger/src/schema/repositories/schema.repository.ts (hunk 4 above)
Comment (bhavanakarwade) [reply apps/ledger/src/schema/repositories/schema.repository.ts:170]: resolved

File: apps/api-gateway/src/schema/interfaces/index.ts (hunk 2 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/schema/interfaces/index.ts:47]: Resolved

**Review by tipusinghaw — APPROVED** [review]

**Pull request merged**